### ISSUE TITLE
Enhancement Idea seeds

### ISSUE THREAD

**Issue by thorwhalen** [i2mint/meshed#2]
# DAG Factories
Make func_nodes (i.e. dag) factory to allow the user to use (or make) ways to efficiently
express how names and therefore connections should be made.

## Connection using type annotations
A function providing`Featurizer` as an output can connect to (i.e. feed) functions that take `Featurizer`s as inputs

## structure through nested lists 
`[..., [a, b, c],...]` meaning we want `a -> b -> c`

# ch_names
Method to change the names (roots, inner, leafs, funcs)

# order
Method to make order compliant with some given order-determining list.
Ordering roots and leafs should make inputs and outputs as wished, but doesn't
carry on to sub-DAGs. Do we need more control on topological sort?

# request-triggered data access

`obj.prop `asks `obj._dag` for the sub-dag to compute
said prop and if a maintained `obj._scope` (the cache) contains sufficient data,
it computes it (and caches it) with minimum computional path.

Basically the pattern I often use lazyprop for.
